### ISSUE TITLE
Deprecation warning about 'punycode' (when running on Node.js v21.1.0)

### ISSUE THREAD

**Issue by EdoardoMolinatti** [node-red/node-red#4424]
### Current Behavior

With the last version of [Node.js](https://nodejs.org/) (v**21.1.0**), when you start **Node-RED** this is the initial output:
```
Welcome to Node-RED
===================

6 Nov 22:21:28 - [info] Node-RED version: v3.1.0
6 Nov 22:21:28 - [info] Node.js  version: v21.1.0
6 Nov 22:21:28 - [info] Windows_NT 10.0.19045 x64 LE
6 Nov 22:21:29 - [info] Loading palette nodes
(node:38340) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
[...]
```
As you can see there's a _deprecation warning_ about the `punycode` module.

From the `punycode` module's [GitHub page](https://github.com/mathiasbynens/punycode.js#installation), I saw that there's a note about the inclusion statement to write when using **Node.js**:

`const punycode = require('punycode/');`
_Note the slash `/` at the end of the require directive string parameter_

I think you should use that include statement in all the **Node-RED** codebase to avoid the warning.
_The explanation about why, it's in the linked GitHub page._

Thank you very much,
regards

### Expected Behavior

The _deprecation warning_ should not show up.

### Steps To Reproduce

Just start Node-RED (using the last version of **Node.js**, v**21.1.0**).

### Example flow

N/A

### Environment

- Node-RED version: v3.1.0
- Node.js version: v21.1.0
- npm version: v10.1.0
- Platform/OS: Windows 10 Pro 64bit (22H2)
- Browser: Google Chrome 118.0.5993.120 (64bit)

**Comment by knolleary** [comment]
Hi @EdoardoMolinatti 

Node-RED doesn't require `punycode` itself. Looking at our dependency tree I can see it is being pulled in by two modules:

```
node-red@3.1.0 /Users/nol/code/node-red/node-red
├─┬ ajv@8.12.0
│ └─┬ uri-js@4.4.1
│   └── punycode@2.1.1 deduped
└─┬ tough-cookie@4.1.3
  └── punycode@2.1.1
```

We'll have to follow up with those modules to get this addressed.

**Comment by EdoardoMolinatti** [comment]
Hi @knolleary,
Thank you for the clarification.
I hope you'll get in touch with the people responsible of [`uri-js`](https://www.npmjs.com/package/uri-js) and [`tough-cookie`](https://www.npmjs.com/package/tough-cookie) then.

**Comment by domdomegg** [comment]
For uri-js there is a PR up to fix this: https://github.com/garycourt/uri-js/pull/95

**Comment by knolleary** [comment]
- `ajv` has released 8.17.1 that removes the dependency on uri-js
 - `tough-cookie` has released 4.1.4 that removes the dependency on punycode

**Comment by knolleary** [comment]
This will be addressed by https://github.com/node-red/node-red/pull/4874
